src/main/sensors/battery.c

```diff
@@ -65,6 +69,13 @@ static uint16_t batteryAdcToVoltage(uint16_t src)
 
 static void updateBatteryVoltage(void)
 {
+    #ifdef USE_ESC_TELEMETRY
+    if (batteryConfig->batteryMeterType == BATTERY_SENSOR_ESC && isEscTelemetryActive()) {
+        vbat = getEscTelemetryVbat();
+        return;
```

Comment:
returning here is problematic, because `vbatLatestADC` that is measured from the on-board ADC below is used for battery stabilisation detection and calculation of number of cells in `updateBattery()`. Ideally we should rewrite this to track non-averaged converted battery voltage either from the on-board ADC or from the ESCs and use this for cell calculation.